tests/schemas/package.py

```diff
@@ -0,0 +1,313 @@
+import datetime
+import unittest
+
+from pydantic import ValidationError
+
+from core import database_arango
+from core.schemas import entity, indicator, observable, package
+
+# from core.schemas.indicator import Indicator
+# from core.schemas.observable import Observable
+from core.schemas.observables import hostname, ipv4
+
+
+class YetiPackageTest(unittest.TestCase):
+    def setUp(self) -> None:
+        database_arango.db.connect(database="yeti_test")
+        database_arango.db.clear()
+
+    def tearDown(self) -> None:
+        database_arango.db.clear()
+
+    def test_package_creation_from_methods(self) -> None:
+        yeti_package = package.YetiPackage(
+            timestamp="2024-04-10T10:00:00Z", source="SecretSource"
+        )
+        yeti_package.add_observable("toto.com", "hostname")
+        yeti_package.add_observable("192.168.1.1", "ipv4")
+        yeti_package.add_entity("Fresh campaign", "campaign")
+        yeti_package.add_indicator(
+            "awesome_regexp", "regex", pattern=".*", diamond="adversary"
+        )
+        yeti_package.add_relationship("Fresh campaign", "toto.com", "contacts")
+        yeti_package.add_relationship(
+            "Fresh campaign", "192.168.1.1", "communicates_with"
+        )
+        yeti_package.save()
+
+        obs1 = observable.Observable.find(value="toto.com", type="hostname")
+        obs2 = observable.Observable.find(value="192.168.1.1", type="ipv4")
+        campaign = entity.Entity.find(name="Fresh campaign", type="campaign")
+        regex = indicator.Indicator.find(name="awesome_regexp", type="regex")
+
+        self.assertIsNotNone(obs1)
+        self.assertIsNotNone(obs2)
+        self.assertIsNotNone(campaign)
+        self.assertIsNotNone(regex)
+
+        vertices, paths, count = campaign.neighbors()
+
+        self.assertEqual(len(paths), 2)
+        self.assertEqual(count, 2)
+        self.assertEqual(len(vertices), 2)
+
+        for path in paths:
+            if path[0].target == obs1.extended_id:
+                self.assertEqual(path[0].source, campaign.extended_id)
+                self.assertEqual(path[0].description, "")
+                self.assertEqual(path[0].type, "contacts")
+            elif path[0].target == obs2.extended_id:
+                self.assertEqual(path[0].source, campaign.extended_id)
+                self.assertEqual(path[0].description, "")
+                self.assertEqual(path[0].type, "communicates_with")
+
+        self.assertIn(obs1.extended_id, vertices)
+        neighbor = vertices[obs1.extended_id]
+        self.assertEqual(neighbor.id, obs1.id)
+        self.assertIn(obs2.extended_id, vertices)
+        neighbor = vertices[obs2.extended_id]
+        self.assertEqual(neighbor.id, obs2.id)
+
+    def test_package_creation_from_dict(self) -> None:
```

Comment:
```suggestion
    def test_package_creation_from_dict_objects(self) -> None:
```

To disambiguate with `test_package_creation_from_json_string`